src/init.cpp

```diff
@@ -1048,6 +1050,40 @@ bool AppInitLockDataDirectory()
     return true;
 }
 
+/**
+ * Once initial block sync is finished and no change in block height or queued downloads,
+ * sync utxo state to protect against data loss
+ */
+static void SyncCoinsTipAfterChainSync(const NodeContext& node)
+{
+    LOCK(node.chainman->GetMutex());
+    if (node.chainman->ActiveChainstate().IsInitialBlockDownload()) {
```

Comment:
Do we want to monitor only the active chainstate? Or should we be paying attention to the background one too?

Reply:
That's a good question.
Please correct me if my assumptions are incorrect, but I believe the other chainstate would be the background syncing while our active is the assumeutxo chainstate which we want to sync quickly and then have active ASAP. In that case, there are only two chainstates, active and background. The background chainstate can take a long time to sync, possibly >24h, which would mean the periodic flush would take effect. So syncing that here would not really be beneficial. For the active chainstate it would be much less than 24h and therefore we would want sync it to disk to protect against data loss.